### PR TITLE
Add mutate method to jsonpath

### PR DESCRIPTION
**Pull request by romange**
Plug it in into OpToggle op.

### CODE REVIEW

File: src/core/json/path.cc [hunk 1]
```diff
@@ -291,6 +340,61 @@ auto Dfs::PerformStep(const PathSegment& segment, const JsonType& node, const Cb
...
+    -> nonstd::expected<void, MatchStatus> {
+  switch (segment.type()) {
+    case SegmentType::IDENTIFIER: {
+      if (!node->is_object())
+        return make_unexpected(MISMATCH);
+
+      auto it = node->find(segment.identifier());
+      if (it != node->object_range().end()) {
+        bool erase = Mutate(cb, it->key(), &it->value());
+        if (erase) {
```
Comment (BorysTheDev) [inline src/core/json/path.cc:353]: nit:
if (Mutate(cb, it->key(), &it->value())) {

File: src/core/json/path.cc [hunk 2]
```diff
@@ -291,6 +340,61 @@ auto Dfs::PerformStep(const PathSegment& segment, const JsonType& node, const Cb
...
+      }
+    } break;
+    case SegmentType::INDEX: {
+      if (!node->is_array())
+        return make_unexpected(MISMATCH);
+      if (segment.index() >= node->size()) {
+        return make_unexpected(OUT_OF_BOUNDS);
+      }
+      bool erase = Mutate(cb, nullopt, &node[segment.index()]);
+      if (erase) {
```
Comment (BorysTheDev) [inline src/core/json/path.cc:365]: if you want to have a name for bool
if (bool erase = Mutate(cb, nullopt, &node[segment.index()]); erase) {

File: src/core/json/path.cc [hunk 3]
```diff
@@ -291,6 +340,61 @@ auto Dfs::PerformStep(const PathSegment& segment, const JsonType& node, const Cb
...
+    case SegmentType::WILDCARD: {
+      if (node->is_object()) {
+        auto it = node->object_range().begin();
+        while (it != node->object_range().end()) {
+          bool res = Mutate(cb, it->key(), &it->value());
+          if (res) {
+            it = node->erase(it);
+          } else {
+            ++it;
+          }
```
Comment (BorysTheDev) [inline src/core/json/path.cc:380]: it = res ? node->erase(it) : it + 1;

File: src/core/json/path.cc [hunk 4]
```diff
@@ -291,6 +340,61 @@ auto Dfs::PerformStep(const PathSegment& segment, const JsonType& node, const Cb
...
+        }
+      } else if (node->is_array()) {
+        auto it = node->array_range().begin();
+        while (it != node->array_range().end()) {
+          bool res = Mutate(cb, nullopt, &*it);
+          if (res) {
+            it = node->erase(it);
+          } else {
+            ++it;
+          }
```
Comment (BorysTheDev) [inline src/core/json/path.cc:390]: it = res ? node->erase(it) : it + 1;

File: src/server/json_family.cc [hunk 5]
```diff
@@ -574,23 +574,49 @@ OpResult<vector<OptSizeT>> OpArrLen(const OpArgs& op_args, string_view key, Json
...
+  } else {
+    auto cb = [&vec](const auto&, JsonType& val) {
+      if (val.is_bool()) {
+        bool current_val = val.as_bool() ^ true;
+        val = current_val;
+        vec.emplace_back(current_val);
+      } else {
+        vec.emplace_back(nullopt);
+      }
+    };
```
Comment (BorysTheDev) [inline src/server/json_family.cc:613]: cb are the same but optional type, I think you can implement one callback via another or do a pointer dereference with constexpr if

**Review by BorysTheDev — APPROVED** [review]

File: src/server/json_family.cc (hunk 5 above)
Comment (romange) [reply src/server/json_family.cc:613]: it's not worth it since it's a transitional code and once we are confident it's robust we will delete the old one

Comment (BorysTheDev) [reply src/server/json_family.cc:613]: any case I have approved it

File: src/core/json/path.cc (hunk 2 above)
Comment (romange) [reply src/core/json/path.cc:365]: removed erase

File: src/core/json/path.cc (hunk 3 above)
Comment (romange) [reply src/core/json/path.cc:380]: nice

**Review by BorysTheDev — APPROVED** [review]

**Review by BorysTheDev — APPROVED** [review]

**Pull request merged**